src/main/scala/com/github/pjfanning/pekkohttpspi/PekkoHttpClient.scala

```diff
@@ -107,11 +106,21 @@ object PekkoHttpClient {
       .getOrElse(ContentTypes.NoContentType)
 
   // This method converts the headers to Akka-http headers and drops content-length and returns content-type separately
-  private[pekkohttpspi] def convertHeaders(headers: java.util.Map[String, java.util.List[String]]): (Option[HttpHeader], immutable.Seq[HttpHeader]) =
-    headers.asScala.foldLeft((Option.empty[HttpHeader], List.empty[HttpHeader])) { case ((ctHeader, hdrs), header) =>
+  private[pekkohttpspi] def convertHeaders(headers: java.util.Map[String, java.util.List[String]]): (Option[HttpHeader], immutable.Seq[HttpHeader]) = {
```

Comment:
This entire function can be made clearer by just using Java mutable collections rather than Scala's `foldLeft` (the usage of `foldLeft` here is slightly confusing) but I am erring on the side of caution here as to note accidentally break anything.

This can always be improved later without breaking bincompat.